### ISSUE TITLE
Problems installing GlabTop2

### ISSUE THREAD

**Issue by Andrea-Vergnano** [WilcoTerink/GlabTop2-py#9]
Hello Wilco,
Thank you for sharing this interesting model.
A few years ago, it worked fine, but now, with all the new versions of Conda and Python my colleagues and I struggle to make it work.
I tried creating a conda virtual environment with python 3.8 instead of 3.9, but it was not enough. The command conda install -c WilcoTerink GlabTop2-py fails to solve the environment, there is always some missing dependency and, as for now, I'm not very practical of conda.

It would be very useful if you update the documentation to specify what the users are supposed to do to make it work, considering that they will download the latest version of anaconda.

THANK YOU!!
Andrea

**Comment by Andrea-Vergnano** [comment]
I add a detail. Today I tried creating an environment with all the packages seen in the documentation, with the correct versions, but it has been hours that it tries to solve the environment...
conda create -n GlabTopENV python=3.8 pcraster=4.3.0 GlabTop2-py -c conda-forge -c WilcoTerink

**Comment by LeldeBry** [comment]
I just installed this model today and it worked so I might be able to help. 
Firstly, conda install or pip install did not work for me either. what worked though was using a combination of most recent full PCRaster installation with all packages (link is in GlabTop2 documentation), only switched to python 3.8 and added geopandas package to the list. then git cloned the GlabTop2 repo (or you can just download it) and put the downloaded main.py, config.cfg, init.py and pycache files in my working directory. then I ran the model commands from conda env where pcraster was installed and it worked. 
Hope it helps!

**Comment by Andrea-Vergnano** [comment]
Dear LeldeBry,
thank you very much for your fast and precise answer. I really appreciated and in these days I tried as you said.

Yes! As you suggested, this command creates the correct environment in which GlabTop2 can run: conda create --name GlabTop2_env -c conda-forge python=3.8 numpy pcraster campo lue matplotlib-base spyder qgis spotpy geopandas

I still had 3 issues, of which I ask help with the 3rd, still unresolved for me. Thank you.

1) I don't know if you had the same issue, but I had to change lines 62-63 of the main.py, because the .dbf files (created with normal QGIS saving as shapefile) came with headers written in capital letters.
So:
 rgi = rgi[['Area','Zmin','Zmax','Slope','Lmax']]
        rgi['dH'] = rgi['Zmax']-rgi['Zmin']
became
 rgi = rgi[['AREA','ZMIN','ZMAX','SLOPE','LMAX']]
        rgi['dH'] = rgi['ZMAX']-rgi['ZMIN']

2) In linux, pcraster seems to fail in reading the correct absolute path. As a temporary working solution, I directly wrote, in the main.py,  self.dem = pcr.readmap("dem.map") at line 71 (and line 75), putting the main.py and the config.cfg in the same folder of all the .map files. I tried other attempts, but I did not understand what was the problem. I installed in windows and this problem was not there.

3) 3rd issue i'm stuck now... I give the "python main.py -i config.cfg" command, it manages to begin the iteration, but it gives this error (both in linux and in windows):


GlabTop2-py is developed by Wilco Terink

Version 2.0.2

Processing glacier: 3 with Tau: 37.77 and hmin: 171.16
	Interpolation run 1
Traceback (most recent call last):
[test.zip](https://github.com/WilcoTerink/GlabTop2-py/files/9749401/test.zip)

  File "main.py", line 314, in <module>
    main(sys.argv[1:])
  File "main.py", line 305, in main
    glabtop.run()
  File "main.py", line 269, in run
    h = self.glacHeights(glacmask, glacdem, glacSlope, innerCells, hmin, tau)
  File "main.py", line 198, in glacHeights
    dH = np.nanmax(tempDem) - np.nanmin(tempDem)
  File "<__array_function__ internals>", line 180, in nanmax
  File "/home/andrea/anaconda3/envs/GlabTop_environment/lib/python3.8/site-packages/numpy/lib/nanfunctions.py", line 476, in nanmax
    res = np.fmax.reduce(a, axis=axis, ou
[test.zip](https://github.com/WilcoTerink/GlabTop2-py/files/9749399/test.zip)
t=out, **kwargs)
ValueError: zero-size array to reduction operation fmax which has no identity


I really don't understand what's happening. Maybe it is a problem of my input files... i attach them here if you have time to give them a look.

**Comment by Andrea-Vergnano** [comment]
The test.zip files i attached went into the code two times, sorry for the mistake
here it is
[test.zip](https://github.com/WilcoTerink/GlabTop2-py/files/9749414/test.zip)

**Comment by Andrea-Vergnano** [comment]
Hello dear developers,
This morning I made some other tests with the data I posted in the previous comment, and I found the problem was in my data. In fact, I was dealing with a glacier situated on the border between France and Italy. And for this reason, the DEM was cut on the border, probably not overextending the size of the glacier in some little parts.

So, now it works. Thanks again for your support.

It would be very helpful, for the next student/scientist who will approach this model, if you put some considerations taken from this github issue into the official documentation, according to me. There is also a very helpful tutorial video on youtube, who may be cited in the documentation (it's in Spanish) https://www.youtube.com/watch?v=14rqwp7ZqwQ

Have a great day
Andrea
